cmd/uplink/cmd/setup_test.go

```diff
@@ -115,3 +121,58 @@ func TestDefaultPortAppliedToSatelliteAddrWithPortColonButNoPort(t *testing.T) {
 	assert.Equal(t, "ahost:7777", setupCmd.Flags().Lookup("satellite-addr").Value.String(),
 		"satellite-addr should contain default port when no port specified")
 }
+
+func TestSaveEncriptionKey(t *testing.T) {
+	expectedKey := make([]byte, rand.Intn(20)+1)
+	_, err := rand.Read(expectedKey)
+	require.NoError(t, err)
+
+	t.Run("ok", func(t *testing.T) {
+		ctx := testcontext.New(t)
+		defer ctx.Cleanup()
+
+		filename := ctx.File("storj-test-cmd-uplink", ".enc.key")
+		err = cmd.SaveEncryptionKey(expectedKey, filename)
+		require.NoError(t, err)
+
+		// Read the key from the file to compare that it matches with the saved one.
+		file, err := os.Open(filename)
+		require.NoError(t, err)
+		defer func() { require.NoError(t, file.Close()) }()
+
+		key := make([]byte, len(expectedKey))
+		_, err = file.Read(key)
+		require.NoError(t, err)
+		assert.Equal(t, expectedKey, key)
+
+		// Check that the key file has a read only file permissions for the user
+		fileInfo, err := file.Stat()
+		require.NoError(t, err)
+		assert.Equal(t, os.FileMode(0400), fileInfo.Mode().Perm())
+	})
+
+	t.Run("error: unexisting dir", func(t *testing.T) {
+		// Create a directory and remove it for making sure that the path doesn't
+		// exist
+		dir, err := ioutil.TempDir("", "storj-test-cmd-uplink")
+		require.NoError(t, err)
+		err = os.RemoveAll(dir)
+		require.NoError(t, err)
+
+		filename := filepath.Join(dir, "enc.key")
+		err = cmd.SaveEncryptionKey(expectedKey, filename)
+		require.Errorf(t, err, "directory path doesn't exist")
+	})
+
+	t.Run("error: file already exists", func(t *testing.T) {
+		// Create an empty file
+		file, err := ioutil.TempFile("", "storj-test-cmd-uplink-key-*")
```

Comment:
Use `testcontext.File` here too.

Reply:
`testcontext.File` doesn't create the file but it doesn't confirm that the file doesn't exist either.

Reply:
`testcontext.File` ensures that it will return the path inside an empty temporary directory. So, unless you have written the file, it won't exist, unless in previous tests you write one.

Reply:
This one requires the file to exists before calling `SaveEncryptionKey`